webui/react/src/pages/ResourcePool/ResourcepoolDetail.tsx

```diff
@@ -62,32 +63,70 @@ const ResourcepoolDetailInner: React.FC = () => {
   const { canManageResourcePoolBindings } = usePermissions();
   const agents = Loadable.getOrElse([], useObservable(clusterStore.agents));
   const resourcePools = useObservable(clusterStore.resourcePools);
+  const navigate = useNavigate();
+  const [canceler] = useState(new AbortController());
+
+  const [tabKey, setTabKey] = useState<TabType>(tab ?? DEFAULT_POOL_TAB_KEY);
+  const [poolsStats, setPoolsStats] = useState<V1RPQueueStat[]>();
+
+  const MAX_USABLE_NODES = 1_000;
+  const MAX_USABLE_SLOTS = 10_000;
+
   const pool = useMemo(() => {
     if (!Loadable.isLoaded(resourcePools)) return;
 
     return resourcePools.data.find((pool) => pool.name === poolname);
   }, [poolname, resourcePools]);
 
+  const totalSlots = useMemo(() => {
+    const total = agents
+      .map((agent) => {
+        let totalSlots = 0;
+        for (const val of Object.values(agent.slotStats.typeStats ?? {})) {
+          totalSlots += val.total;
+        }
+        return totalSlots;
+      })
+      .reduce((a, b) => a + b, 0);
+    return total;
+  }, [agents]);
+
+  const agentsWithSlots = useAsync(async () => {
+    if (agents.length > MAX_USABLE_NODES || totalSlots > MAX_USABLE_SLOTS) {
+      return [];
+    }
+    try {
+      const response = await getAgents({ excludeSlots: false });
+      return response;
```

Comment:
should be able to just do
```suggestion
      return getAgents({ excludeSlots: false });
```

Reply:
fixed